src/infra/flow-system/inmem/src/flow/flow_event_store_inmem.rs

```diff
@@ -343,6 +343,34 @@ impl FlowEventStore for FlowEventStoreInMem {
         Box::pin(futures::stream::iter(flow_ids_page))
     }
 
+    #[tracing::instrument(level = "debug", skip_all)]
+    fn get_all_flow_ids_by_datasets(
+        &self,
+        dataset_ids: Vec<DatasetID>,
+        pagination: FlowPaginationOpts,
+    ) -> (usize, FlowIDStream) {
```

Comment:
I think that you can still separate the "count" query from "list" query, and it will be more effecient.
For "count", we don't have to iterate. We can query `all_flows_by_dataset` for each of the requested datasets and accumulate the lengths of each collection, if anything is found at all.

While for listing, sure, you need to iterate.

Reply:
Removed